src/geometry/manifold/smooth_manifold_with_corners.lean

```diff
@@ -619,6 +619,38 @@ instance prod {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
 
 end smooth_manifold_with_corners
 
+namespace topological_space.opens
+
+open topological_space
+
+variables  {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+  {E : Type*} [normed_group E] [normed_space 𝕜 E]
+  {H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+  {M : Type*} [topological_space M] [charted_space H M] [smooth_manifold_with_corners I M]
+  (s : opens M)
+
+instance : smooth_manifold_with_corners I s :=
+{ ..topological_space.opens.has_groupoid (times_cont_diff_groupoid ∞ I) s }
+
+end topological_space.opens
+
+def homeomorph.smooth_manifold_with_corners {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+  {E : Type*} [normed_group E] [normed_space 𝕜 E]
+  {H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+  {M : Type*} [topological_space M] [charted_space H M] [smooth_manifold_with_corners I M]
+  {M' : Type*} [topological_space M']
+  (e : homeomorph M M') : @smooth_manifold_with_corners 𝕜 _ E _ _ H _ I M' _ (e.charted_space H) :=
+{ compatible := λ f f' hf hf', begin rcases hf with ⟨ϕ, hϕ1, hϕ2⟩,
+    rcases hf' with ⟨ψ, hψ1, hψ2⟩,
+    rw [hϕ2, hψ2, homeomorph.symm_to_local_homeomorph, local_homeomorph.trans_symm_eq_symm_trans_symm,
+    local_homeomorph.symm_symm, local_homeomorph.trans_assoc,
+    ←local_homeomorph.trans_assoc e.to_local_homeomorph, homeomorph.trans_symm_to_local_homeomorph,
+    local_homeomorph.refl_trans],
```

Comment:
If you click the "Files changed" tab now, it shows a message next to the line that's too long:
```suggestion
    rw [hϕ2, hψ2, homeomorph.symm_to_local_homeomorph, 
      local_homeomorph.trans_symm_eq_symm_trans_symm, local_homeomorph.symm_symm, 
      local_homeomorph.trans_assoc, ←local_homeomorph.trans_assoc e.to_local_homeomorph, 
      homeomorph.trans_symm_to_local_homeomorph, local_homeomorph.refl_trans],
```